mmv1/third_party/terraform/services/alloydb/resource_alloydb_secondary_instance_test.go

```diff
@@ -0,0 +1,759 @@
+  reserved_peering_ranges = [google_compute_global_address.private_ip_alloc.name]
+}
+`, context)
+}
+
+// This test passes if we are able to create a secondary instance with maximum fields
+func TestAccAlloydbInstance_secondaryInstanceMaximumFields(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckAlloydbInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAlloydbInstance_secondaryInstanceMaximumFields(context),
+			},
+			{
+				ResourceName:            "google_alloydb_instance.secondary",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"cluster", "instance_id", "reconciling", "update_time", "labels", "terraform_labels"},
+			},
+		},
+	})
+}
+
+func testAccAlloydbInstance_secondaryInstanceMaximumFields(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_alloydb_cluster" "primary" {
+  cluster_id = "tf-test-alloydb-primary-cluster%{random_suffix}"
+  location   = "us-central1"
+  network      = google_compute_network.default.id
+}
+
+resource "google_alloydb_instance" "primary" {
+  cluster       = google_alloydb_cluster.primary.name
+  instance_id   = "tf-test-alloydb-primary-instance%{random_suffix}"
+  instance_type = "PRIMARY"
+
+  machine_config {
+    cpu_count = 2
+  }
+
+  depends_on = [google_service_networking_connection.vpc_connection]
+}
+
+resource "google_alloydb_cluster" "secondary" {
+  cluster_id   = "tf-test-alloydb-secondary-cluster%{random_suffix}"
+  location     = "us-east1"
+  network      = google_compute_network.default.id
+  cluster_type = "SECONDARY"
+
+  continuous_backup_config {
+    enabled = false
+  }
+
+  secondary_config {
+    primary_cluster_name = google_alloydb_cluster.primary.name
+  }
+
+  deletion_policy = "FORCE"
+
+  depends_on = [google_alloydb_instance.primary]
+}
+
+resource "google_alloydb_instance" "secondary" {
+  cluster       = google_alloydb_cluster.secondary.name
+  instance_id   = "tf-test-alloydb-secondary-instance%{random_suffix}"
+  instance_type = google_alloydb_cluster.secondary.cluster_type
+
+  machine_config {
+    cpu_count = 2
+  }
+
+  labels = {
+    test_label = "test-alloydb-label"
+  }
+
+  annotations = {
+    test_annotation = "test-alloydb-annotation"
+  }
+
+  query_insights_config {
+      query_plans_per_minute = 10
+      query_string_length = 2048
+      record_application_tags = true
+      record_client_address = true
+  }
+
+  gce_zone = "us-east1-b"
+
+  availability_type = "REGIONAL"
+
+  depends_on = [google_service_networking_connection.vpc_connection]
+
+  lifecycle {
+    ignore_changes = [
```

Comment:
Can we remove this lifecycle block? If not, why not?

Reply:
The fields `gce_zone` and `annotations` are deprecated and not supported by Alloydb now.
I have removed the fields from the tests, and the life_cycle.ignore_changes field too.
Please let me know if this test needs to be changed in any other way